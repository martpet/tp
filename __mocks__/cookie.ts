export default {
  serialize: vi.fn().mockReturnValue('dummyCookie'),
  parse: vi.fn(),
};
