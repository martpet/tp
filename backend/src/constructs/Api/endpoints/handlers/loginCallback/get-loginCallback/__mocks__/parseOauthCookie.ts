import { OauthCookieProps } from '~/constructs/Api/types';

export const dummyOauthCookieProps: OauthCookieProps = {
  stateNonce: 'dummyState',
  idTokenNonce: 'dummyIdTokenNonce',
  codeVerifier: 'dummyCodeVerifier',
};

export const parseOauthCookie = vi.fn().mockReturnValue(dummyOauthCookieProps);
