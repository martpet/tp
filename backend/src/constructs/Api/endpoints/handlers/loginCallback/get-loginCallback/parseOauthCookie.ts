import {
  APIGatewayProxyEventV2,
  OauthCookieProps,
  parseEventCookies,
} from 'lambda-layer';

export const parseOauthCookie = (event: APIGatewayProxyEventV2) => {
  try {
    const { oauth } = parseEventCookies<'/loginCallback'>(event);
    if (!oauth) throw new Error();
    return JSON.parse(oauth) as Partial<OauthCookieProps>;
  } catch (e) {
    console.error(e);
    return {};
  }
};
