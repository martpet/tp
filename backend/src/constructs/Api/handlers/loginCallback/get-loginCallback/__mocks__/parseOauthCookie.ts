import { OauthCookieProps } from '../../../../apiTypes';

export const dummyOauthCookieProps: OauthCookieProps = {
  stateNonce: 'dummyState',
  idTokenNonce: 'dummyIdTokenNonce',
  codeVerifier: 'dummyCodeVerifier',
};

export const parseOauthCookie = vi.fn().mockReturnValue(dummyOauthCookieProps);
