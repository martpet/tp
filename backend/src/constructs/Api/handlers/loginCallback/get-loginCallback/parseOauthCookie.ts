import cookie from 'cookie';

import { OauthCookieProps } from '../../../apiTypes';
import { cookieName } from '../../../apiUtils';

export const parseOauthCookie = (cookies?: string[]) => {
  try {
    const parsedCookies = cookie.parse(cookies!.join(';'));
    const oauthCookie = parsedCookies[cookieName('oauth')];
    return JSON.parse(oauthCookie) as Partial<OauthCookieProps>;
  } catch (e) {
    console.error(e);
    return {};
  }
};
