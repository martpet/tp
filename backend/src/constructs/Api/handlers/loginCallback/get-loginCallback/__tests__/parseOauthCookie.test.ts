import cookie from 'cookie';

import { parseOauthCookie } from '../parseOauthCookie';

vi.mock('cookie');

const args = [['dummyCookie1', 'dummyCookie2']];

beforeEach(() => {
  vi.mocked(cookie.parse).mockReturnValue({
    oauth: JSON.stringify({ dummyOauthCookieContentKey: 'dummyOauthCookieContentValue' }),
  });
});

describe('parseOauthCookie', () => {
  it('calls "cookie.parse" with correct args', () => {
    parseOauthCookie(...args);
    expect(vi.mocked(cookie.parse).mock.calls).toMatchSnapshot();
  });

  it('returns a correct value', () => {
    expect(parseOauthCookie(...args)).toMatchSnapshot();
  });

  describe('when "cookie.parse" throws', () => {
    beforeEach(() => {
      vi.mocked(cookie.parse).mockImplementation(() => {
        throw new Error();
      });
    });
    it('returns a correct value', () => {
      expect(parseOauthCookie(...args)).toMatchSnapshot();
    });
  });

  describe('when "oauth" cookie is missing', () => {
    beforeEach(() => {
      vi.mocked(cookie.parse).mockReturnValue({});
    });
    it('returns a correct value', () => {
      expect(parseOauthCookie(...args)).toMatchSnapshot();
    });
  });

  describe('when "oauth" cookie content is not json', () => {
    beforeEach(() => {
      vi.mocked(cookie.parse).mockReturnValue({ oauth: 'notJson' });
    });
    it('returns a correct value', () => {
      expect(parseOauthCookie(...args)).toMatchSnapshot();
    });
  });
});
