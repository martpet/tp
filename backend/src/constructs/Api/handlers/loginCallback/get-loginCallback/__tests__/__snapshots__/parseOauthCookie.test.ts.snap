// Vitest Snapshot v1

exports[`parseOauthCookie > calls "cookie.parse" with correct args 1`] = `
[
  [
    "dummyCookie1;dummyCookie2",
  ],
]
`;

exports[`parseOauthCookie > returns a correct value 1`] = `
{
  "dummyOauthCookieContentKey": "dummyOauthCookieContentValue",
}
`;

exports[`parseOauthCookie > when "cookie.parse" throws > returns a correct value 1`] = `{}`;

exports[`parseOauthCookie > when "oauth" cookie content is not json > returns a correct value 1`] = `{}`;

exports[`parseOauthCookie > when "oauth" cookie is missing > returns a correct value 1`] = `{}`;
